vmm_core/virt/src/x86/vp.rs

```diff
@@ -1242,10 +1242,46 @@ impl StateElement<X86PartitionCapabilities, X86VpInfo> for Xss {
 #[repr(C)]
 #[derive(Default, Debug, PartialEq, Eq, Protobuf, Inspect)]
 #[mesh(package = "virt.x86")]
-pub struct CacheControl {
+pub struct Pat {
     #[mesh(1)]
     #[inspect(hex)]
-    pub msr_cr_pat: u64,
+    pub value: u64,
+}
+
+impl HvRegisterState<HvX64RegisterName, 1> for Pat {
+    fn names(&self) -> &'static [HvX64RegisterName; 1] {
+        &[HvX64RegisterName::Pat]
+    }
+
+    fn get_values<'a>(&self, it: impl Iterator<Item = &'a mut HvRegisterValue>) {
+        for (dest, src) in it.zip([self.value]) {
+            *dest = src.into();
+        }
+    }
+
+    fn set_values(&mut self, it: impl Iterator<Item = HvRegisterValue>) {
+        for (src, dest) in it.zip([&mut self.value]) {
+            *dest = src.as_u64();
+        }
+    }
+}
+
+impl StateElement<X86PartitionCapabilities, X86VpInfo> for Pat {
+    fn is_present(_caps: &X86PartitionCapabilities) -> bool {
+        true
+    }
+
+    fn at_reset(_caps: &X86PartitionCapabilities, _vp_info: &X86VpInfo) -> Self {
+        Self {
+            value: X86X_MSR_DEFAULT_PAT,
+        }
+    }
+}
+
+#[repr(C)]
+#[derive(Default, Debug, PartialEq, Eq, Protobuf, Inspect)]
+#[mesh(package = "virt.x86")]
+pub struct CacheControl {
     #[mesh(2)]
```

Comment:
Renumber?

Reply:
Wasn't sure if that'd be a compat hazard somewhere or not.